utils/amt-integration-tests/tests/data_helpers.py

```diff
@@ -0,0 +1,341 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+
+import pandas as pd
+from sqlalchemy import engine, text
+
+from .table_helpers import read_keyvalue_pairs_as_dataframe
+
+
+DESCRIPTOR_SQL = """
+INSERT INTO edfi.Descriptor (CodeValue, ShortDescription, Description, Namespace)
+    VALUES ('<cv>', '<sd>', '<d>', '<n>');
+DECLARE @id int = @@identity;
+INSERT INTO <schema>.<tbl> (<tbl>Id) VALUES (@id);
+"""
+
+SCHEMA_LMSX = "lmsx"
+SCHEMA_EDFI = "edfi"
+APPEND_OPTIONS = {"if_exists": "append", "index": False}
+
+# This static variable helps us keep track of what has already been uploaded so
+# that we can avoid re-uploading, and thus encountering a duplicate key error.
+already_loaded = {}
+
+
+def _get_value(row: pd.Series, column: str) -> str:
+    # Escape any apostrophes so that this is safe to run
+    return row[column].replace("'", "''")
+
+
+def _get_edfi_options(engine: engine.base.Engine) -> dict:
+    return {"con": engine, "schema": SCHEMA_EDFI, **APPEND_OPTIONS}
+
+
+def _get_lmsx_options(engine: engine.base.Engine) -> dict:
+    return {"con": engine, "schema": SCHEMA_LMSX, **APPEND_OPTIONS}
+
+
+def _get_descriptor_id_by_codevalue_and_namespace(
+    engine: engine.base.Engine, codevalue: str, namespace: str
+) -> int:
+    descriptor_id = 0
+    with engine.connect() as connection:
+        sql = text(
+            f"""
+                SELECT DescriptorId FROM edfi.Descriptor
+                WHERE CodeValue = '{codevalue}' and namespace = '{namespace}'"""
+        )
+        result = connection.execute(sql, engine)
+        for row in result:
+            descriptor_id = row["DescriptorId"]
+
+    return descriptor_id
+
+
+def _get_descriptor_id_by_codevalue(
+    engine: engine.base.Engine, codevalue: str
+) -> int:
+    descriptor_id = 0
+    with engine.connect() as connection:
+        sql = text(
+            f"""
+                SELECT top 1 DescriptorId FROM edfi.Descriptor
+                WHERE CodeValue = '{codevalue}'"""
+        )
+        result = connection.execute(sql, engine)
+        for row in result:
+            descriptor_id = row["DescriptorId"]
+
+    return descriptor_id
```

Comment:
I'll stop repeating this logic, will place it in a single method or something to stop having two very similar functions